hudi-hadoop-mr/src/main/java/org/apache/hudi/hadoop/HoodieCopyOnWriteTableInputFormat.java

```diff
@@ -254,6 +260,8 @@ private List<FileStatus> listStatusForSnapshotMode(JobConf job,
               .stream()
               .flatMap(Collection::stream)
               .map(fileSlice -> createFileStatusUnchecked(fileSlice, fileIndex, virtualKeyInfoOpt))
+              .filter(FileStatusOpt -> FileStatusOpt.isPresent())
+              .map(FileStatus -> FileStatus.get())
```

Comment:
Add these lines to avoid IllegalStateException, when reading optimized query to mor with file slice without base file